axum-extra/src/extract/cookie/private.rs

```diff
@@ -102,6 +102,7 @@ use std::{convert::Infallible, fmt, marker::PhantomData};
 ///     }
 /// }
 /// ```
+#[derive(Clone)]
```

Comment:
This will actually derive an impl like this

```rust
impl<K: Clone> Clone for PrivateCookieJar<K> {
    fn clone(&self) -> Self {
        Self {
            jar: self.jar.clone(),
            key: self.key.clone(),
            _marker: self._marker,
        }
    }
}
```

i.e. where `K` must also be `Clone`. But that is not strictly necessary since the struct doesn't actually contain a `K`. It's just references the type via `PhantomData`.

So a more correct implementation would be

```rust
impl<K> Clone for PrivateCookieJar<K> {
    fn clone(&self) -> Self {
        Self {
            jar: self.jar.clone(),
            key: self.key.clone(),
            _marker: self._marker,
        }
    }
}
```

Same goes for `SignedCookieJar`

Reply:
Sorry, got confused by **K = Key**